test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/VersionRange.java

```diff
@@ -7,42 +7,85 @@
  */
 package org.elasticsearch.test.rest.yaml.section;
 
+import org.elasticsearch.Build;
 import org.elasticsearch.Version;
+import org.elasticsearch.common.VersionId;
 import org.elasticsearch.test.VersionUtils;
+import org.elasticsearch.test.rest.ESRestTestCase;
 
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
+import java.util.Set;
 
-public record VersionRange(Version lower, Version upper) {
+public interface VersionRange {
 
-    public boolean contains(Version currentVersion) {
-        return lower != null && upper != null && currentVersion.onOrAfter(lower) && currentVersion.onOrBefore(upper);
-    }
+    boolean matches(Set<String> nodesVersions);
```

Comment:
This definitely could be a `Predicate<Set<String>>`. Either add it as a subinterface, or could use `Predicate<Set<String>>` directly and have a `VersionRanges` helper class?